crates/redis_interface/src/lib.rs

```diff
@@ -138,10 +138,16 @@ impl RedisConnectionPool {
             };
         }
     }
-    pub async fn on_error(&self) {
-        while let Ok(redis_error) = self.pool.on_error().recv().await {
+
+    pub async fn on_error(&self, option_tx: Option<tokio::sync::oneshot::Sender<()>>) {
```

Comment:
- Does this tx need to be optional?
- Please don't prefix variable name with type information, we aren't following this conversion.

Reply:
I removed the Options as the method is called only at one place moreover I refactored the variable name